lib/drivers/abstract/schema.js

```diff
@@ -128,13 +128,74 @@ module.exports = {
     await this.query(sql);
   },
 
+  /**
+   * @param {Object} opts.attributes
+   * @param {Array | boolean} updateOnDuplicate enable 'ON DUPLICATE KEY UPDATE'/'ON CONFLICT DO UPDATE SET' or not
+   * @param {Array | string} opts.uniqueKeys uniqueKeys ON CONFLICT (uniqueKeys) only available in Postgres and SQLite
+   */
+  updateOnDuplicate(opts = {}) {
+    if (!opts.updateOnDuplicate) return null;
+    const { escapeId } = this;
+
+    let uniqueKeys = [];
+    let updateKeys = [];
+
+    const { attributes } = opts;
+
+    if (opts.uniqueKeys) {
+      if (Array.isArray(opts.uniqueKeys) && opts.uniqueKeys.length) {
+        for (const field of opts.uniqueKeys) {
+          const attribute = escapeId(field);
+          uniqueKeys.push(attribute);
+        }
+      } else {
+        uniqueKeys.push(escapeId(opts.uniqueKeys));
+      }
+    } else {
+      // conflict_target must be unique
+      // get all unique keys
+      if (attributes) {
+        for (const key in attributes) {
+          const att = attributes[key];
+          // use the first unique key
+          if (att.unique) {
+            uniqueKeys.push(escapeId(att.columnName));
+            break;
+          }
+        }
+      }
+      const { primaryKey } = opts;
+      if (!uniqueKeys.length) uniqueKeys.push(escapeId(primaryKey));
+      // default use id as primary key
+      if (!uniqueKeys.length) uniqueKeys.push(escapeId('id'));
```

Comment:
Looks like it's possible to push `id` into `uniqueKeys` twice? Will that cause sql error?